perf/perf_metric.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2021 IBM
+# Author: Nageswara R Sastry <rnsastry@linux.ibm.com>
+
+import platform
+from avocado import Test
+from avocado.utils import distro, genio, process
+from avocado.utils.software_manager import SoftwareManager
+
+IS_POWER_NV = 'PowerNV' in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0')
+
+
+class perf_metric(Test):
+
+    """
+    This tests all metric/metric group events
+    :avocado: tags=perf,metric,events
+    """
+    # Initializing fail command list
+    fail_cmd = list()
+
+    def _create_all_metric_events(self, match):
+        cmd = "perf list %s" % match
+        output = process.system_output(cmd, shell=True, ignore_status=True)
+        for ln in output.decode().split('\n'):
+            ln = ln.strip()
+            # Removing empty line, header and comment
```

Comment:
"Skipping  empty.. .. .." would be an apt comment here

Reply:
Changed